core/src/main/java/smile/regression/OLS.java

```diff
@@ -96,6 +97,10 @@
      */
     private double[][] coefficients;
     /**
+     * The gamma matrix for recursive least squares used in online updates
```

Comment:
What exactly is the gamma matrix? Can you add comments how gamma is defined and used?

Reply:
Gamma is the covariance matrix (X'X)^-1. You can find out more about this at https://en.wikipedia.org/wiki/Online_machine_learning#Online_learning:_recursive_least_squares

I decided to used only the diagonal elements for gamma for computational reasons (for SVD we would have to compute the inverse of VSS'V', since X'X=VSS'V'. This can be done using the pseudoinverse) and by some tests I did where the MSE for using the full covariance matrix were very close to using only the singular values from S or the diagonal elments from the cholesky inverse.

I can change gamma to be the full covariance matrix, if you want me to.